src/Microsoft.Extensions.FileProviders.Physical/PhysicalFilesWatcher.cs

```diff
@@ -124,7 +124,7 @@ private IChangeToken GetOrAddFilePathChangeToken(string filePath)
                     new[]
                     {
                         changeToken,
-                        new PollingFileChangeToken(new FileInfo(filePath))
+                        new PollingFileChangeToken(new FileInfo(Path.Combine(_root, filePath)))
```

Comment:
Should we first check `Path.IsPathRooted` first? Looking at the code, it seems we never validate `filePath` is relative or absolute.